lib/models/addon.js

```diff
@@ -365,19 +364,14 @@ let addonProto = {
         return true;
       }
 
-      // If package.json and index.js names do not match, we should check for scoped vs non-scoped
-      const parsed = npa(parentName);
-      let parsedName = parsed.name;
-      if (parsed.scope) {
-        parsedName = parsedName.replace(parsed.scope, '').slice(1);
-      }
-      if (parsedName === this.name) {
-        if (!this._hasWarnedForMismatchedNames) {
-          this.ui.writeWarnLine(
-            'Your names in package.json and index.js should match. ' +
-            `You currently have ${parsed.name} in package.json and ${this.name} in index.js.`);
+      // Addon names in index.js and package.json should be the same at all times whether they have scope or not.
+      if (this.root === this.parent.root) {
+        if (parentName !== this.name) {
+          let pathToDisplay = process.cwd() === this.root ? process.cwd() : path.relative(process.cwd(), this.root);
```

Comment:
To understand correctly, this code is to display the full root, unless contained within the cwd(), then only should the relative path.

Is that worth it, or should we simply just always show the full path?